optimum/utils/input_generators.py

```diff
@@ -1090,11 +1090,12 @@ def __init__(
             random_sequence_length_range=random_sequence_length_range,
             **kwargs,
         )
-        self.num_kv_heads = self.num_kv_heads = (
-            normalized_config.num_kv_heads
-            if (normalized_config.new_decoder_architecture or not normalized_config.multi_query)
-            else 1
-        )
+        if normalized_config.new_decoder_architecture and normalized_config.multi_query:
+            self.num_kv_heads = normalized_config.num_attention_heads
+        elif normalized_config.new_decoder_architecture:
+            self.num_kv_heads = normalized_config.num_kv_heads
+        else:
+            self.num_kv_heads = 1
         self.head_dim = self.hidden_size // self.num_attention_heads
 
```

Comment:
also should be udpated [here](https://github.com/huggingface/optimum/blob/3b5c486821628d1f1e4c34a59d097e5a5f4e1803/optimum/exporters/onnx/model_configs.py#L405) and [here](https://github.com/huggingface/optimum/blob/3b5c486821628d1f1e4c34a59d097e5a5f4e1803/optimum/onnxruntime/modeling_decoder.py#L852) as well

Reply:
also why not have this value set in the `normalized_config` directly and use it in both `FalconDummyPastKeyValuesGenerator` and `ORTFalconForCausalLM` @fxmarty ? (removing the need to have this check in both places)